File: cpp_solutions/1448. Count Good Nodes in Binary Tree.cc
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int find_ans(TreeNode* root, int maxs) {
        int ans = 0;
        if (root->val >= maxs) {
            ++ans;
        }
        if (root->right!=nullptr) {
            ans+= find_ans(root->right, max(maxs, root->val));
        }
        if (root->left!=nullptr) {
            ans+= find_ans(root->left, max(maxs, root->val));
        }
        return ans;
    }
    int goodNodes(TreeNode* root) {
        return find_ans(root,-100000);
    }
};

File: python3_solution_set0/1448. Count Good Nodes in Binary Tree.py
##ss
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        
        self.ans = 0
        self.expand(root,-float('inf'))
        return self.ans
    
    def expand(self,root,path_max):
        #print(root.val,self.ans)
        if root.val >= path_max:
            #print(root.val,"inside")
            self.ans+=1
            path_max = root.val
            if root.left:
                self.expand(root.left,path_max)
                
            if root.right:
                self.expand(root.right,path_max)
            
        else:
            if root.left:
                self.expand(root.left,path_max)
                
            if root.right:
                self.expand(root.right,path_max)
        #print(root.val,self.ans) 
        

            
        
        
